lib/Dialect/FIRRTL/Transforms/EmitOMIR.cpp

```diff
@@ -285,8 +285,7 @@ void EmitOMIRPass::runOnOperation() {
         tracker.nla =
             dyn_cast_or_null<NonLocalAnchor>(symtbl->lookup(nlaSym.getAttr()));
         removeTempNLAs.push_back(tracker.nla);
-      }
-      if (sramIDs.erase(tracker.id))
+      } else if (sramIDs.erase(tracker.id) && !tracker.nla)
```

Comment:
I think the `!tracker.nla` is unnecessary given that this is doing an `else if` now?